src/components/CommonComponents/Pagination/index.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { FC, MouseEvent, useCallback, useMemo } from 'react';
+import { buildPaginationModel } from './util/buildPaginationModel';
+import Page from './Page';
+import styles from './index.module.scss';
+
+interface Props {
+  currentPage: number;
+  hrefBuilder?: (n: number) => string;
+  marginPageCount: number;
+  pageCount: number;
+  onPageChange?: (e: MouseEvent, n: number) => void;
+  showPages?: boolean;
+  surroundingPageCount?: number;
+  wrapperClassName?: string;
+}
+
+const defaultHrefBuilder = (pageNum: number): string => `#${pageNum}`;
+
+const Pagination: FC<Props> = ({
+  pageCount,
+  currentPage,
+  onPageChange,
+  hrefBuilder = defaultHrefBuilder,
+  marginPageCount = 1,
+  showPages = true,
+  surroundingPageCount = 2,
+  wrapperClassName,
+}) => {
+  const pageChange = useCallback(
+    (n: number) => (e: MouseEvent) => onPageChange && onPageChange(e, n),
+    [onPageChange]
+  );
+
+  const model = useMemo(
+    () =>
+      buildPaginationModel(
+        pageCount,
+        currentPage,
+        !!showPages,
+        marginPageCount,
+        surroundingPageCount
+      ),
+    [pageCount, currentPage, showPages, marginPageCount, surroundingPageCount]
+  );
+
+  const pageElements = useMemo(
+    () =>
+      model.map(page => (
+        <li key={`${page.type}${page.num}`}>
+          <Page
```

Comment:
The name `Page` is confusing. Maybe it should be called `PaginationPage` I know it is rendundant, but we could mistake this with an actual page.

Reply:
fixed